app/services/generate_bill_run_summary.service.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+/**
+ * @module GenerateBillRunSummaryService
+ */
+
+const Boom = require('@hapi/boom')
+
+const { BillRunModel } = require('../models')
+
+class GenerateBillRunSummaryService {
+  /**
+  * Initiates the generation of a bill run summary.
+  *
+  * @param
+  * @returns
+  */
+  static async go (billRunId) {
+    const billRun = await BillRunModel.query().findById(billRunId)
+
+    await this._validateBillRun(billRun, billRunId)
+    await this._setGeneratingSummaryStatus(billRun)
+    await this._summariseBillRun(billRun)
+
+    return billRun
+  }
+
+  static _validateBillRun (billRun, billRunId) {
```

Comment:
We should also check if the bill run is `$editable()`. You'll probably have to do this after checking if the status is `generating` because that _should_ be a status that causes `$editable()` to return false but we've not yet included it! 😳